### PR TITLE
feat(kfileupload): introduce kui tokens [KHCP-7706]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Addresses: https://konghq.atlassian.net/browse/KHCP-7706

Changes:
- introduces `kui-` design tokens in `KFileUpload` component
- removes any usage of Kongponents utility classes in `KFileUpload` component template

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [ ] **Tests pass:** check the output of yarn test
* [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [ ] **Framework style:** abides by the essential rules in Vue's style guide
* [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: docs/components/file-upload.md [hunk 1]
```diff
@@ -224,21 +224,21 @@ Specify an icon to display to the left of the `placeholder text` if prop `type`
 
 ### iconSize
 
-The size of the `icon` being displayed (default is `26`).
+The size of the `icon` being displayed (default is `24px`).
```
Comment (adamdehaven) [inline docs/components/file-upload.md:227]: question: why is the size changing from a number to a string?

File: src/components/KFileUpload/KFileUpload.vue [hunk 2]
```diff
@@ -1,11 +1,11 @@
 <template>
   <div
-    class="k-file-upload w-100"
+    class="k-file-upload"
```
Comment (adamdehaven) [inline src/components/KFileUpload/KFileUpload.vue:3]: nit: this can all be on one line now

File: src/components/KFileUpload/KFileUpload.vue [hunk 3]
```diff
@@ -156,19 +157,19 @@ const props = defineProps({
     default: null,
   },
   /**
-     * Set icon size
-     */
+    * Set icon size
+    */
```
Comment (adamdehaven) [inline src/components/KFileUpload/KFileUpload.vue:161]: nit: can you fix indentation in the whole file?

File: src/components/KFileUpload/KFileUpload.vue [hunk 4]
```diff
@@ -156,19 +157,19 @@ const props = defineProps({
...
   iconSize: {
     type: String,
-    default: '26',
+    default: KUI_ICON_SIZE_50,
```
Comment (adamdehaven) [inline src/components/KFileUpload/KFileUpload.vue:164]: issue: I don't believe you can directly reference an imported const in prop declarations.

If I missed this in other PRs, we'll need to fix there as well.

File: src/components/KFileUpload/KFileUpload.vue [hunk 5]
```diff
@@ -214,7 +215,7 @@ const onFileChange = (evt: any): void => {
 
   const fileSize = fileInput?.value[0]?.size
 
-  hasUploadError.value = fileSize > maximumFileSize.value
+  hasUploadError.value = fileSize as Number > maximumFileSize.value
```
Comment (adamdehaven) [inline src/components/KFileUpload/KFileUpload.vue:218]: suggestion
```suggestion
  hasUploadError.value = Number(fileSize) > maximumFileSize.value
```

File: src/components/KFileUpload/KFileUpload.vue [hunk 6]
```diff
@@ -298,43 +313,43 @@ const resetInput = (): void => {
...
+    background-color: var(--kui-color-background-transparent, $kui-color-background-transparent);
     border: none;
     cursor: pointer;
     height: var(--spacing-lg);
-    padding: var(--type-xxs) 6px;
+    padding: $kFileUploadRemoveButtonPadding;
     position: absolute;
-    right: 118px;
-    top: 38px;
+    right: $tmp-spacing-120;
```
Comment (adamdehaven) [inline src/components/KFileUpload/KFileUpload.vue:323]: issue: this is named too similarly to a kui token

question: why is this a temp variable? Seems like it would just remain hard-coded?

**Review by adamdehaven — CHANGES_REQUESTED** [review]

File: src/components/KFileUpload/KFileUpload.vue (hunk 4 above)
Comment (portikM) [reply src/components/KFileUpload/KFileUpload.vue:164]: Why not? It seems to be working fine (maybe need to pull it into consuming app to confirm, is that what you're thinking?)

Also yes, I will fix it the other places

File: src/components/KFileUpload/KFileUpload.vue (hunk 5 above)
Comment (portikM) [reply src/components/KFileUpload/KFileUpload.vue:218]: This doesn't solve the issue, still need to do `as Number` (the error is `Operator '>' cannot be applied to types 'number' and 'Number'`)

File: src/components/KFileUpload/KFileUpload.vue (hunk 6 above)
Comment (portikM) [reply src/components/KFileUpload/KFileUpload.vue:323]: Removed this, left it there accidentally

File: src/components/KFileUpload/KFileUpload.vue (hunk 4 above)
Comment (adamdehaven) [reply src/components/KFileUpload/KFileUpload.vue:162]: Hmm... I saw an error when I did this in the `icons` repository. Let me try again over there and see if I'm misremembering something

**Review by adamdehaven — APPROVED** [review]
